syntax = "proto3";
package persefone;
import "persefone/interfaces/proto/comm.proto";
import "persefone/interfaces/proto/models.proto";
import "persefone/interfaces/proto/data.proto";



service InferenceService{
    rpc ModelsList(DModelRequest) returns (DModelResponse) {}
    rpc ActivateModel(DModelRequest) returns (DModelResponse) {}
    rpc DeactivateModel(DModelRequest) returns (DModelResponse) {}
    rpc Inference(DInferenceRequest) returns (DInferenceResponse) {}
}

message DModelRequest{
    string model_name = 1;
    string model_category = 2;
    bool fetch_data = 3;
}

message DModelResponse{
    ResponseStatus status = 1;
    repeated DModel models = 2;
}

message DInferenceRequest{
    DTensorBundle bundle = 2;
}

message DInferenceResponse{
    ResponseStatus status = 1;
    DTensorBundle bundle = 2;
}